ifrs17/Import/Importers.ipynb

```diff
@@ -1180,13 +1173,52 @@
             "cell_type": "code",
             "source": [
                 "Import.DefineFormat(ImportFormats.DataNodeParameter, async (options, dataSet) => {",
-                "\n    Guid partitionId = new Guid();",
+                "\n    Activity.Start();",
+                "\n    var primaryArgs = GetArgsFromMain(dataSet) with {ImportFormat = ImportFormats.DataNodeParameter};",
+                "\n    primaryArgs.ValidateArgsForPeriod();",
+                "\n    if(ApplicationMessage.HasErrors()) return Activity.Finish();",
                 "\n    var workspace = Workspace.CreateNew();",
-                "\n    var log = await UploadDataNodeParameterToWorkspaceAsync(dataSet, partitionId, workspace, options.TargetDataSource);",
-                "\n    await workspace.CommitToAsync<SingleDataNodeParameter,PartitionByReportingNode>(options.TargetDataSource, partitionId, snapshot: false);",
-                "\n    await workspace.CommitToAsync<InterDataNodeParameter,PartitionByReportingNode>(options.TargetDataSource, partitionId, snapshot: false);    ",
-                "\n    ",
-                "\n    return log;",
+                "\n    workspace.Initialize(x => x.FromSource(options.TargetDataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());",
+                "\n",
+                "\n    var committedParameters = await options.TargetDataSource.Query<DataNodeParameter>().ToArrayAsync();",
+                "\n    var log = await UploadDataNodeParameterToWorkspaceAsync(dataSet, primaryArgs, workspace, options.TargetDataSource); ",
+                "\n",
+                "\n    if(log.Errors.Any()) return Activity.Finish().Merge(log); ",
+                "\n    var toCommitParameters  = (await workspace.Query<DataNodeParameter>().ToArrayAsync()).Except(committedParameters, ParametersComparer.Instance());",
+                "\n    if (!toCommitParameters.Any()) {",
+                "\n        ApplicationMessage.Log(Warning.VariablesAlreadyImported);   ",
+                "\n        return Activity.Finish().Merge(log);",
+                "\n    }",
+                "\n",
+                "\n    var allArgs = await GetAllArgsAsync(primaryArgs, options.TargetDataSource, ImportFormats.DataNodeParameter);",
+                "\n    var targetDataNodes = toCommitParameters.Select(x => x.DataNode).ToHashSet();",
```

Comment:
For InterDataNodeParameter you also have a linkedDataNode. Why is this not a targetDataNode?

Reply:
Because the linked data nodes are retrieved from the secondary scope of the import storage employing the relaxed logic. I think this might be correct, right?

Reply:
You are super right! If I do not consider also the LinkedDataNode and upload them too to the workspace, the import storage wouldn't have a chance to retrieve the corresponding IfrsVariables or RawVariables. Now I fixed this. Thanks!